infoschema/tables.go

```diff
@@ -1266,6 +1266,9 @@ func keyColumnUsageInTable(schema *model.DBInfo, table *model.TableInfo) [][]typ
 		}
 		for i, key := range fk.Cols {
 			col := nameToCol[key.L]
+			if col == nil {
```

Comment:
I prefer not to add this check, this should act as something like assert to expose inconsistency problems.